### ISSUE TITLE
Nushell infinitely consumes memory opening some sqlite files

### ISSUE THREAD

**Issue by noah427** [nushell/nushell#13088]
### Describe the bug

I'm not sure exactly what's happening here or what about this sqlite file causes it, but opening this file with the open command causes nushell to stall and consume all of the ram my system has. I've only had this issue on 0.94.1 this same exact file worked perfectly fine in 0.93 and 0.92.

### How to reproduce

Open the attached sqlite file with `open data.db`
[data.zip](https://github.com/user-attachments/files/15688565/data.zip)

### Expected behavior

Open and display the sqlite file as it had in the past few major versions.

### Screenshots

![image](https://github.com/nushell/nushell/assets/40678304/566607e9-01b5-4805-9175-2131013b0d93)
^ nushell eating my ram

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.94.1                                                            |
| major              | 0                                                                 |
| minor              | 94                                                                |
| patch              | 1                                                                 |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09) (built from a source tarball) |
| cargo_version      | cargo 1.77.1                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| allocator          | mimalloc                                                          |
| features           | default, sqlite, system-clipboard, trash, which                   |
| installed_plugins  |                                                                   |

### Additional context

I'm running nixos and have nushell installed from unstable nixpkgs.

**Comment by devyn** [comment]
Just a hunch based on barely looking at it, but this might actually be something to do with table formatting. Emoji/unicode issue. The following works fine:

```nushell
open data.db | get profiles | save profiles.csv
```

but I can see emoji for sure.

**Comment by fdncred** [comment]
It doesn't like this row. Not sure why exactly.
```json
{
    "id": 134,
    "created_at": "2024-05-13 00:26:33.530673601+00:00",
    "updated_at": "2024-05-13 00:26:33.530673601+00:00",
    "deleted_at": null,
    "about": "口寂しい🚬\nALT, trust me I'm actually happy in all my pictures, I just don't know how to smile 🗣️😔",
    "count_answers": 31,
    "drugs": "None",
    "excercise": "None",
    "has_kids": "None",
    "long_distance": "None",
    "name": "Ray",
    "orientation": "None",
    "religion": "None",
    "star_sign": "None",
    "age": 26,
    "drinking": "None",
    "education": "None",
    "gender": "Man",
    "height": 0,
    "location": "Okazaki, Japan",
    "looking_for": "None",
    "occupation": "None",
    "clubs": "",
    "relationship_status": "None",
    "smoking": "None",
    "wants_kids": "None",
    "dump_index": null
  }
```

**Comment by fdncred** [comment]
One more thing, if your terminal is only wide enough to show 4 columns, the data is rendered fine. However, with my default terminal showing 14 of these columns, it pukes.

@zhiburt Do you have any idea what's going on here? We suspect that the table rendering code is getting confused with this data somehow. Your expertise would be appreciated if you have time to look into this.

**Comment by zhiburt** [comment]
I did not reproduced exactly;

Somewhat I think it's caused by #12786, at least on mine end it ends this way :disappointed: 
If set `open data.sql | table --width=1000000000` then it's good.

_____

According to #12786 yes ....; maybe today will address it.....want say nothing here ..... :sob: (just haven't got time to be honest...)

**Comment by fdncred** [comment]
interesting that `table --width=1000000000` allows it. I'm guessing it is the wrapping algorithm then.

**Comment by zhiburt** [comment]
Soooooooooooooooooooooooo
Super duper fanny thing I've got to find.....

Apparently (well I am confident in it).
there was a breaking change in `unicode_width` crate.

`nushell` fetches `2.1.12`
`tabled` fetches `2.1.11` (and was originally (I guess) tested with)

So here's an issue. (I bet strongly the sole one)

```rust
    #[test]
    fn test_unicode_width() {
        let text = "\u{200d}♀\u{fe0f}";
        // on 0.2.11
        assert_eq!(unicode_width::UnicodeWidthStr::width(text), 1);
        // on 0.2.12
        assert_eq!(unicode_width::UnicodeWidthStr::width(text), 2);
    }
```

What's more important is that on both versions if calculated per char the result is the same `1`.
So I am not sure whether it's a issue with `unicode_width` or by design some feature introduced.
But surely it was not supposed to be released as a minor bump I guess??????
And what's more important it seems like an error to me. 

So yes I'll probably try to communicate it further.
That's just what I've got for this night.
Just keep you posted.

PS: I am not implying anything specific with a comment about version bump; I have some struggles with it myself :smile: :disappointed:

**Comment by Manishearth** [comment]
> But surely it was not supposed to be released as a minor bump I guess??????

Yes, it was, please do not assume that the algorithm in that crate will continue to always produce the same results.

This is not abnormal in the Rust community: as long as usages of the crate still compile and the crate does what it is documented to do, it is typically still considered a non-breaking change unless otherwise documented. unicode-width never promised algorithm stability.

**Comment by 1gmar** [comment]
I've run into this issue as well, and it appears to be reproducible with the file formats that end up converted into a table and are around ~1.2gb large: 

all ram gets consumed, and it either crashes or, after finally rendering the result, about half of the ram remains in use by the nushell process.

Nushell version: 0.108.0
System: Linux (NixOS 26.05)
kernel_version 6.12.59

**Comment by zhiburt** [comment]
Maybe need to revisit it... then?

Of course after #16240 I do still hope to land it 😅

**Comment by fdncred** [comment]
ya, a fix would be great!
